### PR TITLE
[Php80] Skip change UniqueEntity constrainst annotation if Entity attribute exists on AnnotationToAttributeRector

### PR DESCRIPTION
**Pull request by samsonasik**
Given the following code:

```php
use Doctrine\ORM\Mapping\Entity;
use Doctrine\ORM\Mapping\Table;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;

/**
 * @UniqueEntity(fields={"username"})
 * @UniqueEntity(fields={"email"})
 */
#[Entity(repositoryClass: 'UserRepository')]
#[Table(name: 'core_user')]
class User
{
}
```

It changes to :

```diff
-/**
- * @UniqueEntity(fields={"username"})
- * @UniqueEntity(fields={"email"})
- */
 #[Entity(repositoryClass: 'UserRepository')]
 #[Table(name: 'core_user')]
+#[Entity(fields: ['username'])]
+#[Entity(fields: ['email'])]
```

which remove `@UniqueEntity` and add new 2 lines `#[Entity]` attribute. It should be skipped.

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉

File: rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/FixtureAutoImported/skip_unique_entity.php.inc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * @UniqueEntity(fields={"username"})
+ * @UniqueEntity("email")
+ */
+#[Entity(repositoryClass: 'UserRepository')]
+#[Table(name: 'users')]
+class User
+{
+}
+
+?>
```
Comment (TomasVotruba) [inline rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/FixtureAutoImported/skip_unique_entity.php.inc]: IMO to make this test work, there should be `@Entity` in the "before" part of fixture.
Now this file is ignored and skipped, so nothing can be imported here.

Bellow the `'Doctrine\ORM\Mapping\Entity'` is added, so it should be convered here too.

```php
AnnotationToAttributeRector::ANNOTATION_TO_ATTRIBUTE => ValueObjectInliner::inline([
                new AnnotationToAttribute('Doctrine\ORM\Mapping\Entity'),
```

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/FixtureAutoImported/skip_unique_entity.php.inc]: this fixture is for run the rector after Entity attribute exists, I will add one more fixture for non-exists Entity attribute

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/FixtureAutoImported/skip_unique_entity.php.inc]: I renamed fixture to `skip_unique_entity_when_entity_attribute_exists.php.inc` https://github.com/rectorphp/rector-src/pull/747/commits/da627e1c78597b21669626f2d8b0d063437ea18f

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/FixtureAutoImported/skip_unique_entity.php.inc]: I added more fixture for do not remove `@UniqueEntity` when `#Entity` attribute not exists yet, with defined `@Entity` annotation so only change `@Entity` annotation to `#Entity` attribute for it.

https://github.com/rectorphp/rector-src/pull/747/commits/69a49cff215c44134e86fe294a7ccadeae83e253

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Pull request merged**